frontend/src/components/collab/CodeEditorPanel.tsx

```diff
@@ -16,18 +18,48 @@ const CodeEditorPanel: FC<CodeEditorPanelProps> = ({
   partner,
   language,
   questionTitle,
-  roomId,
+  roomId
 }) => {
-  const [defaultCode, setDefaultCode] = useState<string>(
-    getCodeTemplate(language, questionTitle)
+
+  const [socketService, setSocketService] = useState<SocketService | null>(null);
+  const [isSocketConnected, setIsSocketConnected] = useState<boolean>(false);
+  const editorRef = useRef(null);
+
+  const [currentCode, setCurrentCode] = useState<string>(
+    getCodeTemplate(language, questionTitle) 
   );
 
-  const handleResetToDefaultCode = () => {
-    setDefaultCode(getCodeTemplate(language, questionTitle));
+  useEffect(() => {
+    if (socketService) {
+      socketService.receiveCodeUpdate(setCurrentCode);
+    }
+  });
+
+  setInterval(() => {
+    if (socketService) {
+      setIsSocketConnected(socketService.getConnectionStatus());
+    }
+  })
```

Comment:
would this cause performance issue?

Reply:
I'm not certain actually... i dont notice anything but maybe ill set it to poll every 0.5/1s instead